### ISSUE TITLE
please include patches for support for PPS via GPIO

### ISSUE THREAD

**Issue by ghost** [raspberrypi/linux#288]
Please also include the PPS-over-GPIO patch: http://ntpi.openchaos.org/pps_pi/ pps-gpio patch for the bcm2708 platform and clone the raspberry kernel tree: https://raw.github.com/lampeh/rpi-misc/master/linux-pps/linux-rpi-pps-gpio-bcm2708.diff

This enables one to sync the RPI using a GPS with PPS (pulse per second) signal.
With that, a raspberry pi can be synced with microsecond accuracy (well, better than what we currently get with internet-sync) via e.g. an NTP receiver.
For example a regular (intel based) pc gives this after only less than a minute of ntp runtime:
#      remote           refid      st t when poll reach   delay   offset  jitter

o127.127.20.0    .GPS.            0 l    -   16   17    0.000    0.959   1.480

**Comment by popcornmix** [comment]
The problem with that patch is it unconditionally assigns GPIO 18 for this purpose. Most people won't want that.
I need a suggested patch that is a no-op when not requested (e.g. through a command line option).
A PR should be submitted, and ideally some evidence that people want this (e.g. please post here if you will make use of this feature).

**Comment by popcornmix** [comment]
Duplicate of https://github.com/raspberrypi/linux/issues/109

**Comment by MozisII** [comment]
I've prepared a patch with a module parameter for the GPIO pin assignment but I haven't yet tested it. I can propose it if somebody validate it.

**Comment by ghost** [comment]
@MozisII

> I've prepared a patch with a module parameter for the GPIO pin
> assignment but I haven't yet tested it. I can propose it if somebody
> validate it.

I've written a line or two of code, so I can do it from a source code
point of view (am still waiting on a package with a gps etc I ordered 5
weeks ago...).

**Comment by ghost** [comment]
For people who want the let their raspberry pi sync with ntp against a pps source (e.g. a gps (adafruit has real nice ones)) WITHOUT patching the kernel, I wrote a small program: http://www.vanheusden.com/time/rpi_gpio_ntp/
In my test, it reached < 0.01us jitter in 3 hours. Offset also somewhat near 0.01us.

**Comment by jera-sea** [comment]
That patch no longer works with the latest kernel version only hunk 2 succeeds.
